Finish rating

tae898 opened the pull request:
* Specify metrics, if possible
* List answer choices, if possible
* Add one more prompt (NLG)
* Make the prompt names more explicit.

The three aspects, _meaning_, _fluency_, and _simplicity_, are rated separately, by the dataset authors.
AFAIK, in daily English, these three words overlap in meaning, so it wasn't trivial to distinguish them when I write them out.
Also, I'm not a native English speaker.

Please let me know if you want me to change anything.

tae898 commented:
I'm trying to check the failed check but can't see the full error message:

```sh
=========================== short test summary info ============================
FAILED test/test_templates.py::test_dataset[dataset204] - ValueError: Templat...
================== 1 failed, 269 passed, 2 warnings in 54.00s ==================
```

I wonder what's happening after `...` ... haha

tae898 commented:
Also it seems that there is something wrong with fetching the dataset?

```sh
Traceback (most recent call last):
  File "test/show_templates.py", line 25, in <module>
    dataset = get_dataset(dataset_name, subset_name)
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/promptsource/utils.py", line 49, in get_dataset
    builder_instance.download_and_prepare()
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datasets/builder.py", line 596, in download_and_prepare
    dl_manager=dl_manager, verify_infos=verify_infos, **download_and_prepare_kwargs
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datasets/builder.py", line 666, in _download_and_prepare
    self.info.download_checksums, dl_manager.get_recorded_sizes_checksums(), "dataset source files"
  File "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/datasets/utils/info_utils.py", line 33, in verify_checksums
    raise ExpectedMoreDownloadedFiles(str(set(expected_checksums) - set(recorded_checksums)))
datasets.utils.info_utils.ExpectedMoreDownloadedFiles: {'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.3', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.1', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.valid.simp.6', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.7', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.4', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.valid.simp.7', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.6', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.8', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.valid.simp.8', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.5', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.0', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.9', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.valid.simp.5', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.test.simp.2', 'https://github.com/facebookresearch/asset/raw/master/dataset/asset.valid.simp.9'}
```

Btw as @VictorSanh pointed out, the promptsource app can only fetch this dataset correctly, at the moment, when you install the master branch of [the huggingface datasets repo](https://github.com/huggingface/datasets).

VictorSanh commented:
Could you try to run `show_new_templates` locally?
The test will continue to fail until there is an official release of datasets...
Check_new_templates doesn't try to pull out the data from the cloud, it's only testing the templates

tae898 commented:
```sh
(dev-python3.7) ➜  promptsource git:(asset-ratings) ✗ python test/show_templates.py promptsource/templates/asset/ratings/templates.yaml
/home/tk/repos/promptsource/promptsource/utils.py:36: FutureWarning: prepare_module is deprecated and will be removed in the next major version of datasets. Use dataset_module_factory or metric_module_factory instead.
  module_path = datasets.load.prepare_module(path, dataset=True)
Reusing dataset asset (/home/tk/.cache/huggingface/datasets/asset/ratings/1.0.0/a1ebd31e2a43bb6d4b5826423c73e8397d1696526af6c99f20da612f51799a8f)
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 195.95it/s]
DATASET  promptsource/templates/asset/ratings/templates.yaml
TEMPLATE
NAME: generate-text-based-on-rating
Is Original Task:  False
{% if rating > 50 %}
{% set label = 1 %}
{% else %}
{% set label = 0 %}
{% endif %}
{% if label == 1 %}
{% set questions= [ "Rewrite the following sentence so that it conveys the information better.",  "Rewrite the following sentence so that it is more fluent.", "Rewrite the following sentence so that it is simpler."] %}
{% else %}
{% set questions= [ "Rewrite the following sentence so that it conveys the information more poorly.",  "Rewrite the following sentence so that it is less fluent.", "Rewrite the following sentence so that it is more complicated."] %}
{% endif %}
{{questions[aspect]}}

{{original}}
|||
{{simplification}}


TEMPLATE
NAME: rate-binary
Is Original Task:  True
{% if rating > 50 %}
{% set label = 1 %}
{% else %}
{% set label = 0 %}
{% endif %}
{% set questions= [ "Does the second sentence better convey the information?",  "Is the second sentence more fluent?", "Is the second sentence simpler?"] %}

First sentence: {{original}}

Second sentence: {{simplification}}

{{questions[aspect]}}. Please answer Yes or No. 
|||
{{answer_choices[label]}}


TEMPLATE
NAME: rate-regression
Is Original Task:  True
{% set statements= [ "the second sentence expresses the underlying meaning the best.",  "the second sentence is more fluent.", "the second sentence is simpler."] %}

First sentence: {{original}}

Second sentence: {{simplification}}

Rate the following statement from 0 (strongly disagree) to 100 (strongly agree): {{statements[aspect]}} 

|||
{{rating}}

TEMPLATE
NAME: rate-regression-simplicity
Is Original Task:  True
First sentence: {{original}}

Second sentence: {{simplification}}

I am scoring these simplification exercises. How easier to read is the second sentence on a scale from 0 (harder to read) to 100 (easier to read)?

|||

{{rating}}


TEMPLATE
NAME: generate-text-based-on-rating
REFERENCE: 
--------

{% if rating > 50 %}
{% set label = 1 %}
{% else %}
{% set label = 0 %}
{% endif %}
{% if label == 1 %}
{% set questions= [ "Rewrite the following sentence so that it conveys the information better.",  "Rewrite the following sentence so that it is more fluent.", "Rewrite the following sentence so that it is simpler."] %}
{% else %}
{% set questions= [ "Rewrite the following sentence so that it conveys the information more poorly.",  "Rewrite the following sentence so that it is less fluent.", "Rewrite the following sentence so that it is more complicated."] %}
{% endif %}
{{questions[aspect]}}

{{original}}
|||
{{simplification}}


        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 7, 'rating': 55}
        --------

        Prompt | X
         Rewrite the following sentence so that it conveys the information better.

        Since
         2000, the recipient of the Kate Greenaway medal has also been presented with the
         Colin Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 5, 'rating': 59}
        --------

        Prompt | X
         Rewrite the following sentence so that it conveys the information better.

        Since
         2000, the recipient of the Kate Greenaway medal has also been presented with the
         Colin Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 5, 'rating': 27}
        --------

        Prompt | X
         Rewrite the following sentence so that it is less fluent.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 3, 'rating': 100}
        --------

        Prompt | X
         Rewrite the following sentence so that it is more fluent.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 2, 'worker_id': 8, 'rating': 36}
        --------

        Prompt | X
         Rewrite the following sentence so that it is more complicated.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 15, 'rating': 31}
        --------

        Prompt | X
         Rewrite the following sentence so that it is less fluent.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 6, 'rating': 49}
        --------

        Prompt | X
         Rewrite the following sentence so that it conveys the information more poorly.
         Since 2000, the recipient of the Kate Greenaway medal has also been presented
         with the Colin Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 23, 'rating': 10}
        --------

        Prompt | X
         Rewrite the following sentence so that it is less fluent.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 29, 'rating': 93}
        --------

        Prompt | X
         Rewrite the following sentence so that it is more fluent.

        Since 2000, the
         recipient of the Kate Greenaway medal has also been presented with the Colin
         Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 29, 'rating': 73}
        --------

        Prompt | X
         Rewrite the following sentence so that it conveys the information better.

        Since
         2000, the recipient of the Kate Greenaway medal has also been presented with the
         Colin Mears award to the value of £5000.

        Y
         Since 2000, the winner of the Kate Greenaway medal has also been given to the
         Colin Mears award of the Kate Greenaway medal.

TEMPLATE
NAME: rate-binary
REFERENCE: Taking questions from the original paper, we use rating to establish a binary classification problem
--------

{% if rating > 50 %}
{% set label = 1 %}
{% else %}
{% set label = 0 %}
{% endif %}
{% set questions= [ "Does the second sentence better convey the information?",  "Is the second sentence more fluent?", "Is the second sentence simpler?"] %}

First sentence: {{original}}

Second sentence: {{simplification}}

{{questions[aspect]}}. Please answer Yes or No. 
|||
{{answer_choices[label]}}


        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 7, 'rating': 55}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Does the second sentence
         better convey the information?. Please answer Yes or No.

        Y
         Yes
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 5, 'rating': 59}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Does the second sentence
         better convey the information?. Please answer Yes or No.

        Y
         Yes
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 5, 'rating': 27}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         more fluent?. Please answer Yes or No.

        Y
         No
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 3, 'rating': 100}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         more fluent?. Please answer Yes or No.

        Y
         Yes
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 2, 'worker_id': 8, 'rating': 36}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         simpler?. Please answer Yes or No.

        Y
         No
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 15, 'rating': 31}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         more fluent?. Please answer Yes or No.

        Y
         No
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 6, 'rating': 49}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Does the second sentence
         better convey the information?. Please answer Yes or No.

        Y
         No
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 23, 'rating': 10}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         more fluent?. Please answer Yes or No.

        Y
         No
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 29, 'rating': 93}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Is the second sentence
         more fluent?. Please answer Yes or No.

        Y
         Yes
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 29, 'rating': 73}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Does the second sentence
         better convey the information?. Please answer Yes or No.

        Y
         Yes

TEMPLATE
NAME: rate-regression
REFERENCE: Require the model to output the rating. This is a regression task whose range is from 0 to 100.
--------

{% set statements= [ "the second sentence expresses the underlying meaning the best.",  "the second sentence is more fluent.", "the second sentence is simpler."] %}

First sentence: {{original}}

Second sentence: {{simplification}}

Rate the following statement from 0 (strongly disagree) to 100 (strongly agree): {{statements[aspect]}} 

|||
{{rating}}

        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 7, 'rating': 55}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence expresses the underlying meaning the best.

        Y
         55
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 5, 'rating': 59}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence expresses the underlying meaning the best.

        Y
         59
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 5, 'rating': 27}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is more fluent.

        Y
         27
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 3, 'rating': 100}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is more fluent.

        Y
         100
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 2, 'worker_id': 8, 'rating': 36}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is simpler.

        Y
         36
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 15, 'rating': 31}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is more fluent.

        Y
         31
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 6, 'rating': 49}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence expresses the underlying meaning the best.

        Y
         49
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 23, 'rating': 10}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is more fluent.

        Y
         10
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 29, 'rating': 93}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence is more fluent.

        Y
         93
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 29, 'rating': 73}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        Rate the following
         statement from 0 (strongly disagree) to 100 (strongly agree): the second
         sentence expresses the underlying meaning the best.

        Y
         73

TEMPLATE
NAME: rate-regression-simplicity
REFERENCE: Prompt model to rate how simplified the sentence is in the general sense, instead of an particular aspect. This is a regression task whose range is from 0 to 100.
--------

First sentence: {{original}}

Second sentence: {{simplification}}

I am scoring these simplification exercises. How easier to read is the second sentence on a scale from 0 (harder to read) to 100 (easier to read)?

|||

{{rating}}

        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 7, 'rating': 55}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         55
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 5, 'rating': 59}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         59
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 5, 'rating': 27}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         27
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 3, 'rating': 100}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         100
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 2, 'worker_id': 8, 'rating': 36}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         36
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 15, 'rating': 31}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         31
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 6, 'rating': 49}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         49
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 23, 'rating': 10}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         10
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 1, 'worker_id': 29, 'rating': 93}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         93
        --------
        Split  full
        Example  {'original': 'Since 2000, the recipient of the Kate Greenaway medal has also been presented with the Colin Mears award to the value of £5000.', 'simplification': 'Since 2000, the winner of the Kate Greenaway medal has also been given to the Colin Mears award of the Kate Greenaway medal.', 'original_sentence_id': 7, 'aspect': 0, 'worker_id': 29, 'rating': 73}
        --------

        Prompt | X
         First sentence: Since 2000, the recipient of the Kate Greenaway medal has also
         been presented with the Colin Mears award to the value of £5000.

        Second
         sentence: Since 2000, the winner of the Kate Greenaway medal has also been given
         to the Colin Mears award of the Kate Greenaway medal.

        I am scoring these
         simplification exercises. How easier to read is the second sentence on a scale
         from 0 (harder to read) to 100 (easier to read)?

        Y
         73
```

tae898 commented:
```sh
python test/test_templates.py promptsource/templates/asset/ratings/templates.yaml
```
This doesn't fail on my local machine, though.

VictorSanh commented:
> This doesn't fail on my local machine, though.

great! it means that the test passed (this test is mainly trying to apply the templates to the few first examples of the dataset and print the resulting output).

The CI will always fail because it's not installing hf dataset from source.

only CI test that is failing is `check_templates` -> `ValueError: Template for dataset asset/ratings with uuid 7dd6e8b6-eae0-40c5-aa5e-1cc24357d85d has unrecognized variable questions`

tae898 commented:
> > This doesn't fail on my local machine, though.
> 
> great! it means that the test passed (this test is mainly trying to apply the templates to the few first examples of the dataset and print the resulting output).
> 
> The CI will always fail because it's not installing hf dataset from source.
> 
> only CI test that is failing is `check_templates` -> `ValueError: Template for dataset asset/ratings with uuid 7dd6e8b6-eae0-40c5-aa5e-1cc24357d85d has unrecognized variable questions` It's failing because you need to define variables outside of conditional statements (more details in the last point of the [contribution guideline](https://github.com/bigscience-workshop/promptsource/blob/main/CONTRIBUTING.md#best-practices))

thanks for the heads-up!

VictorSanh commented:
lgtm! thank you @tae898 
merging now

the test `show_new_templates` on master too.
